### PR TITLE
feat(plasma-web/hope): Handling dropdown menu height

### PR DESCRIPTION
**Pull request by Yakutoc**
Необходимо добавить возможность управлять высотой и scroll-ом выпадающего меню. 

Добавлен новый **API**. 

Новые `props` для компонента `Dropdown`: 

- **listOverflow**
- **listHeight**

```jsx
<Dropdown                 
   listHeight={6}
   listOverflow="scroll" 
/>
```

Или в компоненте `DropdownList` переопределить значения для css custom properties:

- **--plasma-dropdown-height**
- **--plasma-dropdown-overflow**

Достаточно обернуть в styled:

```jsx
    const StyledDropdownList = styled(DropdownList)`
        --plasma-dropdown-height: 6rem;
        --plasma-dropdown-overflow: scroll
    `
```
  
**Было:**
Снимок экрана 2023-03-14 в 10 27 33

**Стало:**
Снимок экрана 2023-03-14 в 10 26 41

-----------

:baby_chick: Download canary assets:

[colors--canary.407.4415463726.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.407.4415463726.xml)  
[color_metro_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.407.4415463726.swift)  
[color_metro_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.407.4415463726.kt)  
[color_metro_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.407.4415463726.ts)  
[color_sberHealth_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.407.4415463726.swift)  
[color_sberHealth_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.407.4415463726.kt)  
[color_sberHealth_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.407.4415463726.ts)  
[color_sbermarket_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.407.4415463726.swift)  
[color_sbermarket_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.407.4415463726.kt)  
[color_sbermarket_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.407.4415463726.ts)  
[color_sberprime_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.407.4415463726.swift)  
[color_sberprime_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.407.4415463726.kt)  
[color_sberprime_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.407.4415463726.ts)  
[color_selgros_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.407.4415463726.swift)  
[color_selgros_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.407.4415463726.kt)  
[color_selgros_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.407.4415463726.ts)  
[color_smbusiness_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.407.4415463726.swift)  
[color_smbusiness_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.407.4415463726.kt)  
[color_smbusiness_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.407.4415463726.ts)  
[PlasmaTokensColor--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.407.4415463726.swift)  
[shadow_sbermarket_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.407.4415463726.ts)  
[typo_mage_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.407.4415463726.swift)  
[typo_mage_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.407.4415463726.kt)  
[typo_mage_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.407.4415463726.ts)  
[typo_plasma_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.407.4415463726.swift)  
[typo_plasma_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.407.4415463726.kt)  
[typo_plasma_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.407.4415463726.ts)  
[typo_ruler_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.407.4415463726.swift)  
[typo_ruler_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.407.4415463726.kt)  
[typo_ruler_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.407.4415463726.ts)  
[typo_sage_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.407.4415463726.swift)  
[typo_sage_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.407.4415463726.kt)  
[typo_sage_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.407.4415463726.ts)  
[typo_sbermarket_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.407.4415463726.swift)  
[typo_sbermarket_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.407.4415463726.kt)  
[typo_sbermarket_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.407.4415463726.ts)  
[typo_soulmate_ios-swift--canary.407.4415463726.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.407.4415463726.swift)  
[typo_soulmate_kotlin--canary.407.4415463726.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.407.4415463726.kt)  
[typo_soulmate_react-native--canary.407.4415463726.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.407.4415463726.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.158.0-canary.407.4415463726.0
  npm install @salutejs/plasma-hope@0.16.0-canary.407.4415463726.0
  npm install @salutejs/plasma-web@1.182.0-canary.407.4415463726.0
  # or 
  yarn add @salutejs/plasma-b2c@1.158.0-canary.407.4415463726.0
  yarn add @salutejs/plasma-hope@0.16.0-canary.407.4415463726.0
  yarn add @salutejs/plasma-web@1.182.0-canary.407.4415463726.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx [hunk 1]
```diff
@@ -1,19 +1,49 @@
...
+
+    box-sizing: border-box;
+    overflow-y: var(--plasma-dropdown-overflow, initial);
 `;
+
+export interface DropdownListCssProperties {
+    /**
+     * Значение css overflow для выпадающего меню.
+     */
+    dropdownListOverflow?: CSSProperties['overflow'];
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:31]: а почему не просто overflow и height ?

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
+    /**
+     * Значение css height для выпадающего меню.
+     * @example dropdownListHeight="11" или dropdownListHeight={11}
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: может стоит написать `2rem` ?
и где-то надо написать что если высота не задается то она равна контенту списка

Comment (Yeti-or) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: и кажется поле example у нас не парсится, но я бы так и оставил

File: website/plasma-web-docs/docs/components/Dropdown.mdx [hunk 3]
```diff
@@ -136,6 +136,86 @@ export function App() {
...
+                dropdownListOverflow="scroll"
+            >
+                <Button text="Опции" />
+            </Dropdown>
+        </section>
+    );
+}
+```
+
+Или переопределить в компоненте `DropdownList` значения для css custom properties:
```
Comment (Yeti-or) [inline website/plasma-web-docs/docs/components/Dropdown.mdx:172]: а давай это останется частью внутреннего API, чем больше возможностей тем больше нам поддерживать потом

File: website/plasma-web-docs/docs/components/Dropdown.mdx [hunk 4]
```diff
@@ -136,6 +136,86 @@ export function App() {
 }
 ```
 
+### Управление высотой выпадающего меню
+Что бы задать высоту и `scroll` для выпадающего меню можно использовать **props** для компонента `Dropdown`:
+- **dropdownListOverflow**
+- **dropdownListHeight**
+
+#### Пример
```
Comment (Yeti-or) [inline website/plasma-web-docs/docs/components/Dropdown.mdx:144]: в сторибук будем добавлять кнобы для высоты?

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: Screenshot 2023-03-14 at 12 18 13

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:31]: Что бы было **точно** понятно к чему именно относиться это свойство.   

@Yeti-or @neretin-trike переименовать?

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: > может стоит написать 2rem ?
и где-то надо написать что если высота не задается то она равна контенту списка

Объясни по другому пожалуйста.

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:31]: да, тоже думаю, что лучш overflow и height оставить, в принципе из контекста самого компонента должно быть понятно, что это к выпадающему списку

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 3 above)
Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:172]: Тут исходил из того что у нас пользователь может сам собрать свой Dropdown и тогда ему нужен механизм управления. 

Props здесь используются на родительском уровне, поэтому решил что уместно указать и такой способ взаимодействия.

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 4 above)
Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:144]: Хороший вопрос, но у нас в `Dropdown` story вообще нет никаких controls для взаимодействия.

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:31]: У нас тот же APi будет и для Select и там уже не совсем понятно к чему именно относятся данные свойства.

`<Select overflow ={} height={} />`

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: что будет если не задавать это значение? понятно ли это из документации?

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 3 above)
Comment (Yeti-or) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:172]: собрать свой дропдаун? в том плане что у нас все состовляющие его торчат наружу?

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 4 above)
Comment (Yeti-or) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:144]: ну давай тогда отдельную задачу заведем про то что в сторе нет кнобов для дропдауна

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 3 above)
Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:172]: Ага, в документации целый раздел про это с примером.  

```md
## Пользовательская сборка
Выпадающие списки допустимо собирать вручную с помощью составных компонентов. Для это этого используйте следующие компоненты:

### DropdownPopup
<Description name="DropdownPopup" />
<PropsTable name="DropdownPopup" />

### DropdownList
<Description name="DropdownList" />

### DropdownItem
<Description name="DropdownItem" />
<PropsTable name="DropdownItem" />
```

File: website/plasma-web-docs/docs/components/Dropdown.mdx (hunk 4 above)
Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Dropdown.mdx:144]: Когда будем объединять компоненты, тогда можно и сторей заняться. 

В рамках одной задачи.

File: packages/plasma-hope/src/components/Dropdown/DropdownList.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-hope/src/components/Dropdown/DropdownList.tsx:34]: >  понятно ли это из документации?

Good point! Допишу в документации этот момент.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-407/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-407/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-407/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-407/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-407/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-407/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-407/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Comment by Yeti-or** [comment]
докинуть в описание пр про новые пропы

**Review by neretin-trike — APPROVED** [review]

**Comment by neretin-trike** [comment]
ништяк :shipit:

**Pull request merged**